From afa37eacb9bf1e71184e0615dcf6e7fff94ccd3b Mon Sep 17 00:00:00 2001
From: John Andersen <john.s.andersen@intel.com>
Date: Mon, 14 Sep 2020 10:59:32 -0700
Subject: [PATCH 09/11] X86: uapi: Add kernel_info struct

In perpetration for using kernel_info to check for existence
paravirtualized control register pinning support, add the structure to
the userspace API headers.

The structure currently only includes the fixed length fields, as no
variable length fields currently exist.

Signed-off-by: John Andersen <john.s.andersen@intel.com>
---
 arch/x86/include/uapi/asm/bootparam.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/x86/include/uapi/asm/bootparam.h b/arch/x86/include/uapi/asm/bootparam.h
index 600a141c8805..665ad78123d5 100644
--- a/arch/x86/include/uapi/asm/bootparam.h
+++ b/arch/x86/include/uapi/asm/bootparam.h
@@ -104,6 +104,13 @@ struct setup_header {
 	__u32	kernel_info_offset;
 } __attribute__((packed));
 
+struct kernel_info {
+	__u32	header;
+	__u32	size;
+	__u32	size_total;
+	__u32	setup_type_max;
+} __attribute__((packed));
+
 struct sys_desc_table {
 	__u16 length;
 	__u8  table[14];
-- 
2.21.0

